docs/www/coprocessors.md

```diff
@@ -0,0 +1,122 @@
+# coprocessors
+
+The coprocessor implementation allows you to deploy functional units that operate on the data moving through redpanda logs. The advantage of using the built in coprocessor engine is a drastic improvement in performance, reducing the raft chatter that would otherwise exist if you were to perform transformations off cluster, and it removes any operational complexity that would be introduced by needed to host/manage your transforms on seperate hardware.
+
+For now you can only deploy Javascript functions, however in the future we plan to support all of the WASM supported languages.
+
+## Getting started
+
+To start you'll need to ensure you have `nodejs`, `rpk` and `redpanda` installed on your machine. Then in your `redpanda.yaml` configuration file, you'll need to enable developer and coprocessing mode like so:
+
+```
+redpanda:
+  ...
+  developer_mode: true
+  enable_coproc: true
+  ...
+```
+
+### Creating your first coprocessor
+
+To start, run `rpk wasm generate` in a fresh directory. This will write an example project scaffolding that you can use to get started even quicker.
+
+```
+.
+└── wasm
+    ├── package.json
+    ├── src
+    │   └── wasm.js
+    ├── test
+    │   └── wasm.test.js
+    └── webpack.js
+```
+
+Your source code will be within the `src` directory. Below is an explaination of the API but for now, there is a default operator in there that will consume from one topic called `produce` and produce this onto another topic called `result` while applying a transformation that just uppercases all characters detected on stream.
+
+Once your done modifying the source, cd into the `wasm` directory and run `npm install`. You'll need to do this when you modify your package dependencies. Then run `npm run build` to package up the finished product. You'll see this in the `dist/` folder, with the same name as the source file:
+
+```
+wasm/dist
+└── wasm.js
+```
+
+### Building & launching the WASM engine
```

Comment:
Ok, I'm a little lost, do I need to have the full source code and vtools to get WASM working?

Reply:
Not if we package the engine, which I think is already done. However this document was more for internal usage or anyone working with the source tree.